data_pipelines_cli/cli_commands/compile.py

```diff
@@ -72,8 +67,32 @@ def _copy_dbt_manifest() -> None:
 
 
 def _replace_k8s_settings(docker_args: DockerArgs) -> None:
-    k8s_config: pathlib.Path = BUILD_DIR.joinpath("dag", "config", "base", "k8s.yml")
-    _replace_image_tag(k8s_config, docker_args)
+    k8s_config = BUILD_DIR.joinpath("dag", "config", "base", "k8s.yml")
+    echo_info(
+        f"Replacing {IMAGE_TAG_TO_REPLACE} with commit SHA = {docker_args.commit_sha}"
+    )
+    replace(k8s_config, IMAGE_TAG_TO_REPLACE, docker_args.commit_sha)
+
+
+def _replace_datahub_with_jinja_vars(env: str) -> None:
+    datahub_config_path: pathlib.Path = BUILD_DIR.joinpath(
+        "dag", "config", "base", "datahub.yml"
+    )
+
+    if not datahub_config_path.exists():
+        echo_warning(
+            f"File config/base/datahub.yml does not exist in {BUILD_DIR}. "
+            "Content will not be replaced."
+        )
+        return
+
+    echo_info(f"Replacing Jinja variables in {datahub_config_path}.")
+    with open(datahub_config_path, "r") as datahub_config_file:
+        updated_config = replace_vars_with_values(
+            yaml.safe_load(datahub_config_file), read_dbt_vars_from_configs(env)
+        )
+    with open(datahub_config_path, "w") as datahub_config_file:
+        yaml.dump(updated_config, datahub_config_file)
```

Comment:
We shoudn't replace config in config dir, we should place it into build dir and take it from there durring deployment.